### DISCUSSION TITLE
Should I always use atomsWithQuery with Suspense? or is there any escape hatch?

### DISCUSSION THREAD

**Discussion by dante01yoon** [pmndrs/jotai#1584] [Q&A]
Greetings, 
 I want to wrap server response with atom, so I used atomsWithQuery from `jotai-tanstack-query`

```tsx
eport const useServerValues = (id: string, scope?: string) => {
  const serverValuesAtoms = useMemo(() => {
    return atomsWithQuery(() => {
      return {
        queryKey: ["SERVER", id],
        queryFn: async ({ queryKey: [_, id] }) => {
          const data = await fetchSomething({ name: id });
          return data
        },
      };
    });
  }, [id]);
  const [dataAtom] = serverValuesAtoms;

  return useAtom(dataAtom);
};

```

while I use this hooks inside component which not wrapped by React.Suspense,  shows error below.

![스크린샷 2022-11-18 오후 1 41 25](https://user-images.githubusercontent.com/6510430/202618495-027d6f7a-0f33-4b63-b502-b14cc511788b.png)
this error normally caused by fetching some data needs wrapped by suspense, 

so,  can't I use atomsWithQuery  with "suspense: false" ?   or  use this hook without  using Suspense ? 

Or should I just use AscnyAtom ?  ( But.. then I can't use cache... )

**Comment by dai-shi** [comment]
```js
const [, statusAtom] = atomsWithQuery(...)
```
is for non-suspense usage. The status includes data 😁 .

**Reply by dante01yoon** [reply]
thank you👍 
I always appreiciate @dai-shi  san's quick response! 

oops! I misunderstood statusAtom includes queryInfos!  

- so suspense options means nothing?  
![스크린샷 2022-11-18 오후 2 04 11](https://user-images.githubusercontent.com/6510430/202622925-40613237-2caa-429c-a40e-804eca737113.png)

---
if I don't need cache, just need wrapping response with atom,  is using AsyncAtom is best practice? 
I think set explicitly atom with response in render phase is cumbersome.

**Reply by dai-shi** [reply]
Yeah, `suspense` option in tanstack/query would be meaningless for us.
Right, if you don't need cache, async atoms would just work.
I'm also experimenting [jotai-cache](https://github.com/jotai-labs/jotai-cache).
